### PR TITLE
Move benchmark apps to extension/benchmark dir

### PR DESCRIPTION
**Pull request by guangy10**

### CODE REVIEW

**Comment by guangy10** [comment]
Resolve linter errors by adding the .clang-format for apple code.

**Comment by guangy10** [comment]
Fixed `validate-android-test-spec`. The job name "Upload AWS Device Farm Android test specs" must be passed as a str.

**Review by huydhn — APPROVED** [review]
Stamped!

**Comment by guangy10** [comment]
More fixes in the `extension/benchmark/apple/Benchmark/README.md`

File: extension/benchmark/apple/Benchmark/Tests/Tests.xcconfig
Comment (guangy10) [inline extension/benchmark/apple/Benchmark/Tests/Tests.xcconfig:1]: > $(SRCROOT)/../../../.. \

@shoumikhin I see some relative paths like this in the project. It looks like these need to be updated as well? Should it point to the `executorch` root dir?

File: extension/benchmark/apple/Benchmark/Tests/Tests.xcconfig [hunk 1]
```diff
@@ -16,7 +16,7 @@ OTHER_LDFLAGS[sdk=iphonesimulator*] = $(inherited) \
 -force_load $(BUILT_PRODUCTS_DIR)/libkernels_quantized-simulator-release.a
 
 HEADER_SEARCH_PATHS[sdk=iphoneos*] = $(inherited) \
-$(SRCROOT)/../../../.. \
+$(SRCROOT)/../../../../.. \
```
Comment (guangy10) [inline extension/benchmark/apple/Benchmark/Tests/Tests.xcconfig:19]: @shoumikhin The relative path should be corrected right? Are there any other relative paths for this app?

Comment (shoumikhin) [reply extension/benchmark/apple/Benchmark/Tests/Tests.xcconfig:19]: Yes, that's right

**Comment by guangy10** [comment]
More relative paths fix in `Benchmark.xcodeproj/project.pbxproj`

**Pull request closed without merging**